src/StimulusBundle/src/Dto/StimulusAttributes.php

```diff
@@ -22,14 +22,32 @@
  */
 class StimulusAttributes implements \Stringable, \IteratorAggregate
 {
+    /**
+     * TODO: To be removed in the next major version.
+     */
+    private bool $useOptimizedRendering;
+
     private array $attributes = [];
 
     private array $controllers = [];
     private array $actions = [];
     private array $targets = [];
 
-    public function __construct(private Environment $env)
+    /**
+     * TODO: Remove the second argument and all related code in the next major version.
+     */
+    public function __construct(private Environment $env /* , bool $useOptimizedRendering = false */)
```

Comment:
no need to do that magic about new argument. PHP does not enforce having a consistent constructor signature in child classes, so adding a new optional constructor argument is not a BC break.

Reply:
Oh, you're right, thanks